### ISSUE TITLE
Add `pad` function

### ISSUE THREAD

**Issue by AlexandreBrown** [Kotlin/multik#149]
I would like to know how to pad using `createAlignedNDArray`.  
From my understanding this function takes a list of lists and makes every list the max length and pads the remaining with zeros or whatever value you want (for each dim).  
![image](https://user-images.githubusercontent.com/26939775/201745770-81a1cb4a-2d8f-4d03-9be0-f205d56499ea.png)
  
Example:  
Let `P` be the size of each dims we want our final matrix to be with padding.  
Let `W` and `H` be 2 sizes (different) that are less than `P`.  
For the sake of this example let's say we have `W`< `H` < `P`.  
Let `a` be an mk array of shape `(W, H)`.
Then applying `createAlignedNDArray(a.toListD3(), 0.0)` would give us a matrix of shape `(H, H)` padded with zeros.  
  
While the name of the function seems correct, I'm interested in knowing if it's possible to use this function to transoform `a` into a matrix of size `(P, P)` instead.  
  
Something like, we provide an mk array and can specify the desired padding for each dimensions.  
See https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html  
  
If it's possible with `createAlignedNDArray` then feel free to let me know.

**Comment by devcrocod** [comment]
In your example where `a` is shaped `(W, H)`, using `createAlignedNDArray(a.toListD3(), 0.0)` will give the same shape (W, H). ndarray in multik is aligned, i.e. all sizes at dimension 0 will be the same. Unfortunately, it will not work to get the shape `(P, P)`. This can be done by converting the output lists to a mutableList and expanding one of them to size P.

I had a request for this function, having figured it out, I called it `createAlignedNDArray`, since this is not a full-fledged `pad`.

In that case, I'll try to add the classic `pad`.

**Comment by AlexandreBrown** [comment]
@devcrocod Awesome, here is my naive implementation for if it until it is released (in case it can help someone) :  
```kotlin
override fun pad(unPaddedImageCHW: MultiArray<Float, D3>, padding: Padding): MultiArray<Float, D3> {

	val outputHeight = unPaddedImageCHW.height + padding.top + padding.bottom

	val outputWidth = unPaddedImageCHW.width + padding.left + padding.right

	return mk.d3arrayIndices(unPaddedImageCHW.channels, outputHeight, outputWidth) { c, y, x ->
		if (isInPaddingArea(x = x, y = y, padding, outputHeight = outputHeight, outputWidth = outputWidth)) {
			0.0f
		} else {
			getImageData(channel = c, x = x, y = y, padding, unPaddedImageCHW)
		}
	}
}

private fun isInPaddingArea(
	x: Int,
	y: Int,
	padding: Padding,
	outputHeight: Int,
	outputWidth: Int,
) =
	x < padding.left ||
			y < padding.top ||
			outputWidth - x <= padding.right ||
			outputHeight - y <= padding.bottom

private fun getImageData(
	channel: Int,
	x: Int,
	y: Int,
	padding: Padding,
	unPaddedImage: MultiArray<Float, D3>,
): Float {
	val regionX = x - padding.left
	val regionY = y - padding.top
	return unPaddedImage[channel][regionY][regionX]
}
```
